addon/index.js

```diff
@@ -60,7 +60,7 @@ let RouterScrollMixin = Mixin.create({
 
     let preserveScrollPosition;
     if (gte('3.6.0-beta.1')) {
-      preserveScrollPosition = getWithDefault(transition, 'intent.router.currentRouteInfos', []).some((routeInfo) => get(routeInfo, 'route.controller.preserveScrollPosition'));
+      preserveScrollPosition = getWithDefault(transition, 'router.currentRouteInfos', []).some((routeInfo) => get(routeInfo, 'route.controller.preserveScrollPosition'));
```

Comment:
btw is `intent` documented anywhere?

Reply:
[`TransitionIntent`](https://github.com/tildeio/router.js/blob/853ba3dda85ca10ceb93ab65618728a9aebfd8ca/lib/router/transition-intent.ts) and [`Transition`](https://github.com/tildeio/router.js/blob/853ba3dda85ca10ceb93ab65618728a9aebfd8ca/lib/router/transition.ts#L47) both come from [router.js](https://github.com/tildeio/router.js/). 

There is an [architecture overview](https://github.com/tildeio/router.js/blob/853ba3dda85ca10ceb93ab65618728a9aebfd8ca/ARCHITECTURE.md#transitionintent), but it's quite dated (2014!) so it might be better to just look under the hood. 🙂

Reply:
So effectively `intent.router` or `router` are one in the same?

Reply:
Yeah, `intent.router` and `router` are the same - the app's router.

Though, I don't see any reason why `intent.router` could be undefined (TS types don't consider that possibility), so that might be a bug in ember. Anyway, for now this fix is the simplest way to get `preserveScrollPosition` working.